worlds/yoshisisland/Regions.py

```diff
@@ -0,0 +1,225 @@
+            {'4-Extra': lambda state: state.has_any({'Extra Panels', 'Extra 4'}, player)})
+        multiworld.get_region('World 5', player).add_exits(['5-Extra'],
+            {'5-Extra': lambda state: state.has_any({'Extra Panels', 'Extra 5'}, player)})
+        multiworld.get_region('World 6', player).add_exits(['6-Extra'],
+            {'6-Extra': lambda state: state.has_any({'Extra Panels', 'Extra 6'}, player)})
+
+    if world.options.minigame_checks.value >= 2:
+        multiworld.get_region('World 1', player).add_exits(['1-Bonus'],
+            {'1-Bonus': lambda state: state.has_any({'Bonus Panels', 'Bonus 1'}, player)})
+        multiworld.get_region('World 2', player).add_exits(['2-Bonus'],
+            {'2-Bonus': lambda state: state.has_any({'Bonus Panels', 'Bonus 2'}, player)})
+        multiworld.get_region('World 3', player).add_exits(['3-Bonus'],
+            {'3-Bonus': lambda state: state.has_any({'Bonus Panels', 'Bonus 3'}, player)})
+        multiworld.get_region('World 4', player).add_exits(['4-Bonus'],
+            {'4-Bonus': lambda state: state.has_any({'Bonus Panels', 'Bonus 4'}, player)})
+        multiworld.get_region('World 5', player).add_exits(['5-Bonus'],
+            {'5-Bonus': lambda state: state.has_any({'Bonus Panels', 'Bonus 5'}, player)})
+        multiworld.get_region('World 6', player).add_exits(['6-Bonus'],
+            {'6-Bonus': lambda state: state.has_any({'Bonus Panels', 'Bonus 6'}, player)})
+
+def create_location(player: int, location_data: LocationData, region: Region, location_cache: List[Location]) -> Location:
+    location = YILocation(player, location_data.name, location_data.code, region)
+    location.access_rule = location_data.rule
+    location.LevelID = location_data.LevelID
+
+    if id is None:
+        location.event = True
+        location.locked = True
+
+    location_cache.append(location)
+
+    return location
+
+def create_region(multiworld: MultiWorld, player: int, locations_per_region: Dict[str, List[LocationData]], location_cache: List[Location], name: str) -> Region:
+    region = Region(name, player, multiworld)
+    region.world = multiworld
+
+    if name in locations_per_region:
+        for location_data in locations_per_region[name]:
+            location = create_location(player, location_data, region, location_cache)
+            region.locations.append(location)
+
+    return region
+
+def connectStartingRegion(multiworld: MultiWorld, player: int, world):
+    menu = multiworld.get_region('Menu', player)
+    world_main = multiworld.get_region('Overworld', player)
+
+    starting_region = multiworld.get_region(f'World {world.options.starting_world.value + 1}', player)
+
+    load_file = Entrance(player, 'Overworld', menu)
+    load_file.connect(world_main)
+    menu.connect(world_main, "Start Game")
+
+    starting_world = Entrance(player, 'Overworld', world_main)
+    starting_world.connect(starting_region)
+    world_main.connect(starting_region, "Overworld")

```

Comment:
```suggestion
def connect_starting_region(world: "YIWorld") -> None:
    menu = world.get_region("Menu")
    world_main = world.get_region("Overworld", player)
    starting_region = world.get_region(f"World {world.options.starting_world + 1}")

    menu.connect(world_main, "Start Game")
    world_main.connect(starting_region)
```

Reply:
Will this work as an apworld in 0.4.4? I'm pretty sure those world.gets didn't exist until recently, but I may be misremembering.